openforcefield/utils/toolkits.py

```diff
@@ -3365,6 +3365,79 @@ def generate_conformers(
         for conformer in molecule2._conformers:
             molecule._add_conformer(conformer)
 
+    def assign_partial_charges(
+        self,
+        molecule,
+        partial_charge_method=None,
+        use_conformers=None,
+        strict_n_conformers=False,
+        _cls=None,
+    ):
+        """
+        Compute partial charges with RDKit, and assign
+        the new values to the partial_charges attribute.
+
+        .. warning :: This API is experimental and subject to change.
+
+        Parameters
+        ----------
+        molecule : openforcefield.topology.Molecule
+            Molecule for which partial charges are to be computed
+        partial_charge_method : str, optional, default=None
+            The charge model to use. One of ['mmff94']. If None, 'mmff94' will be used.
+
+            * 'mmff94': Applies partial charges using the Merck Molecular Force Field
+                        (MMFF). This method does not make use of conformers, and hence
+                        ``use_conformers`` and ``strict_n_conformers`` will not impact
+                        the partial charges produced.
+        use_conformers : iterable of simtk.unit.Quantity-wrapped numpy arrays, each with shape (n_atoms, 3) and dimension of distance. Optional, default = None
+            Coordinates to use for partial charge calculation. If None, an appropriate number of conformers will be generated.
+        strict_n_conformers : bool, default=False
+            Whether to raise an exception if an invalid number of conformers is provided for the given charge method.
+            If this is False and an invalid number of conformers is found, a warning will be raised.
+        _cls : class
+            Molecule constructor
+
+        Raises
+        ------
+        ChargeMethodUnavailableError if the requested charge method can not be handled by this toolkit
+
+        ChargeCalculationError if the charge method is supported by this toolkit, but fails
+        """
+
+        import numpy as np
+        from rdkit.Chem import AllChem
+
+        SUPPORTED_CHARGE_METHODS = {"mmff94"}
+
+        if partial_charge_method is None:
+            partial_charge_method = "mmff94"
+
+        partial_charge_method = partial_charge_method.lower()
+
+        if partial_charge_method not in SUPPORTED_CHARGE_METHODS:
+            raise ChargeMethodUnavailableError(
+                f"partial_charge_method '{partial_charge_method}' is not available from RDKitToolkitWrapper. "
+                f"Available charge methods are {list(SUPPORTED_CHARGE_METHODS)} "
+            )
+
```

Comment:
Sorry for the post-merge comments. I think this PR is nearly perfect, but it was weird to see `use_conformers` go untouched here. 

To mirror the behavior of other charge-generation methods, I think we should have a sanity check for "the user asked for mmff94 but also tried to force the use of specific conformers so something is wrong here". There's a lot of indirection that happens with ToolkitRegistry delegation and exception handling, so this kind of check has raised important red flags when things go wrong deep down in the stack. 

I suggest:
```
        mol_copy = Molecule(molecule)
        if use_conformers is not None:
            mol_copy._conformers = None
            for conformer in use_conformers:
                mol_copy._add_conformer(conformer)
            self._check_n_conformers(
                mol_copy,
                partial_charge_method=partial_charge_method,
                min_confs=0,
                max_confs=0,
                strict_n_conformers=strict_n_conformers,
            )
        rdmit_molecule = mol_copy.to_rdkit()
```

and then `rdkit_molecule = mol_copy.to_rdkit()` 

Also, we should update the releasehistory with this great new feature :-)